barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_trace.hpp

```diff
@@ -339,19 +338,26 @@ class AvmTraceBuilder {
     // Mapping of side effect counter -> value
     ExecutionHints execution_hints;
 
+    template <IntermRegister R>
```

Comment:
IIUC IntermRegister is an enum, and this template can just be a parameter. If possible, then prefer avoiding templates.